jgrapht-core/src/main/java/org/jgrapht/alg/interfaces/MatchingAlgorithm.java

```diff
@@ -80,9 +90,28 @@
         Set<E> getEdges();
 
         /**
-         * Returns an iterator over the edges in the matching.
-         * @return iterator over the edges in the matching.
+         * Returns true if vertex v is incident to an edge in this matching.
+         * @param v vertex
+         * @return true if vertex v is incident to an edge in this matching.
          */
+        default boolean isMatched(V v){
+            return getGraph().edgesOf(v).stream().anyMatch(getEdges()::contains);
```

Comment:
I'm not sure. I've cached the result of `getEdges()` just in case. Thanks for the suggestion. 
I don't think that this default interface implementation will be used often though. The default implementation is added to ofter a certain level of backward compatibility. In most cases, `MatchingImpl` will be used, which implements this function much more efficiently.